src/main/java/tutorlink/command/AddComponentCommand.java

```diff
@@ -27,11 +27,10 @@ public CommandResult execute(AppState appState, HashMap<String, String> hashmap)
 
         int weightage = convertWeightageToValidInt(weightageNumber);
 
-        if((weightage + Component.totalWeight) > MAX_WEIGHT) {
+        int totalWeight = weightage + appState.components.getTotalWeighting();
+        if ((totalWeight) > MAX_WEIGHT) {
             throw new InvalidWeightingException(String.format(Commons.ERROR_INVALID_TOTAL_WEIGHTING,
-                    Component.totalWeight + weightage));
-        } else {
-            Component.totalWeight += weightage;
+                    totalWeight));
         }
```

Comment:
Computing totalWeight every time significantly reduces the code but also adds overhead. 

However for the sake of defensibility I think this is acceptable.